analyzers/src/SonarAnalyzer.Common/SymbolicExecution/Roslyn/OperationProcessors/ArithmeticCalculator.cs

```diff
@@ -167,7 +167,8 @@ private static NumberConstraint CalculateRemainder(NumberConstraint left, Number
         // If the absolute value of the divisor is bigger than the absolute value of the dividend, the result is equal to the dividend =>
         // If the absolute value of every divisor is bigger than the absolute value of any dividend => resulting range == dividend range.
         // Otherwise, the result is between 0 and the absolute value of the divisor - 1 for positive dividends or in the same range multiplied by -1 for negative dividends.
-        if (left.CanBeNegative || BigInteger.Abs(left.Max.Value) < MinOfAbsoluteValues(right))
+        if (left.CanBeNegative
+            || (left.Max is { } && BigInteger.Abs(left.Max.Value) < MinOfAbsoluteValues(right)))
```

Comment:
Compared to the original code, this is really hard to read now. I'd go back to how it was before this PR.
If you want to keep it this way, please change `left.Max is { }` to `left.Max is not null`, though.

Reply:
I was waiting for this comment actually - I think that it because too overengineered I'll revert.